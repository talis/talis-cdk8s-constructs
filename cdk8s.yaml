language: typescript
imports:
  - k8s@1.30.0
