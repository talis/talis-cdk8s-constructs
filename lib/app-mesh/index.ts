export * from "./ingress-gateway-nlb";
export * from "./app-service";
