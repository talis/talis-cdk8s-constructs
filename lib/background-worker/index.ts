export * from "./background-worker-props";
export * from "./background-worker";
