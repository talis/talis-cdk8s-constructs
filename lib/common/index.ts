export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
