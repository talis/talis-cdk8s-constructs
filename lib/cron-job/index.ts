export * from "./cron-job-props";
export * from "./cron-job";
