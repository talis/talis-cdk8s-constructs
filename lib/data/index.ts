export * from "./config-map";
export * from "./secret";
export * from "./factory";
