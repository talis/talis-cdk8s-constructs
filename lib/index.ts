export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
