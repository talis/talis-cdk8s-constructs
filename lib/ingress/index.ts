export * from "./ingress-props";
export * from "./ingress";
