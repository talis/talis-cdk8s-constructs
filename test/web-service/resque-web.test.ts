import { Testing } from "cdk8s";
import { ResqueWeb } from "../../lib";
import { IntOrString } from "../../imports/k8s";

describe("ResqueWeb", () => {
  test("Creates resque web objects", () => {
    const chart = Testing.chart();
    new ResqueWeb(chart, "resque-web", {
      externalUrl: "http://resque.example.com",
    });
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });

  test("Customizations", () => {
    const chart = Testing.chart();
    new ResqueWeb(chart, "resque-web", {
      externalUrl: "https://resque-web.example.com",
      tlsDomain: "*.example.com",
      release: "latest",
      env: [
        { name: "RAILS_ENV", value: "production" },
        { name: "RAILS_RESQUE_REDIS", value: "redis:6379" },
        {
          name: "SECRET_KEY_BASE",
          valueFrom: {
            secretKeyRef: {
              key: "SECRET_KEY_BASE",
              name: "resque-web-secrets",
            },
          },
        },
      ],
      externalHostname: "resque-web.example.com",
    });
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });

  test("selectorLabels are merged", () => {
    const chart = Testing.chart();
    new ResqueWeb(chart, "resque-web", {
      externalUrl: "http://resque.example.com",
      tlsDomain: "*.example.com",
      selectorLabels: {
        test: "true",
      },
    });
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });

  test("ingressAnnotations are merged", () => {
    const chart = Testing.chart();
    new ResqueWeb(chart, "resque-web", {
      externalUrl: "http://resque.example.com",
      tlsDomain: "*.example.com",
      ingressAnnotations: {
        "talis.io/foo": "bar",
      },
    });
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });

  test("It does not include podDisruptionBudget by default", () => {
    const chart = Testing.chart();
    new ResqueWeb(chart, "resque-web", {
      externalUrl: "http://resque.example.com",
    });
    const results = Testing.synth(chart);
    const pdbs = results.filter((obj) => obj.kind === "PodDisruptionBudget");
    expect(pdbs).toHaveLength(0);
  });

  test("Allows specifying podDisruptionBudget", () => {
    const chart = Testing.chart();
    new ResqueWeb(chart, "resque-web", {
      externalUrl: "http://resque.example.com",
      replicas: 2,
      podDisruptionBudget: {
        maxUnavailable: IntOrString.fromNumber(1),
      },
    });
    const results = Testing.synth(chart);
    const pdbs = results.filter((obj) => obj.kind === "PodDisruptionBudget");
    expect(pdbs).toHaveLength(1);
  });
});
