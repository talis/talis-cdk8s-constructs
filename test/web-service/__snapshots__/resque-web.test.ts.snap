// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResqueWeb Creates resque web objects 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "annotations": Object {
        "talis.io/chat": "https://talis.slack.com/archives/C04P9DPCX",
        "talis.io/description": "Frontend to the Resque job queue system",
        "talis.io/eks-dashboard": "None",
        "talis.io/graphs": "None",
        "talis.io/incidents": "None",
        "talis.io/issues": "https://github.com/talis/platform/issues",
        "talis.io/logs": "None",
        "talis.io/repository": "https://github.com/talis/resque-web-container",
        "talis.io/runbook": "None",
        "talis.io/uptime": "None",
        "talis.io/url": "http://resque.example.com",
      },
      "labels": Object {
        "app": "resque",
        "instance": "resque-web",
        "release": "stable",
        "role": "server",
      },
      "name": "test-resque-web-resque-web-service-c81bf306",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 3000,
          "protocol": "TCP",
          "targetPort": 3000,
        },
      ],
      "selector": Object {
        "app": "resque",
        "instance": "resque-web",
        "role": "server",
      },
      "type": "NodePort",
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1",
    "kind": "Ingress",
    "metadata": Object {
      "annotations": Object {
        "alb.ingress.kubernetes.io/listen-ports": "[{\\"HTTP\\":80}]",
        "alb.ingress.kubernetes.io/load-balancer-attributes": "idle_timeout.timeout_seconds=60",
        "alb.ingress.kubernetes.io/load-balancer-name": "resque-web",
        "alb.ingress.kubernetes.io/success-codes": "200,303",
        "alb.ingress.kubernetes.io/tags": "service=resque,instance=resque-web",
        "alb.ingress.kubernetes.io/target-type": "instance",
      },
      "labels": Object {
        "app": "resque",
        "instance": "resque-web",
        "release": "stable",
        "role": "server",
      },
      "name": "test-resque-web-resque-web-ingress-c8c81984",
    },
    "spec": Object {
      "defaultBackend": Object {
        "service": Object {
          "name": "test-resque-web-resque-web-service-c81bf306",
          "port": Object {
            "number": 3000,
          },
        },
      },
      "ingressClassName": "aws-load-balancer-internal",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "resque",
        "instance": "resque-web",
        "release": "stable",
        "role": "server",
      },
      "name": "test-resque-web-c82b282b",
    },
    "spec": Object {
      "replicas": 1,
      "revisionHistoryLimit": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "resque",
          "instance": "resque-web",
          "role": "server",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "resque",
            "instance": "resque-web",
            "release": "stable",
            "role": "server",
          },
        },
        "spec": Object {
          "affinity": Object {
            "podAntiAffinity": Object {
              "preferredDuringSchedulingIgnoredDuringExecution": Array [
                Object {
                  "podAffinityTerm": Object {
                    "labelSelector": Object {
                      "matchLabels": Object {
                        "app": "resque",
                        "instance": "resque-web",
                        "role": "server",
                      },
                    },
                    "topologyKey": "topology.kubernetes.io/zone",
                  },
                  "weight": 100,
                },
              ],
            },
          },
          "automountServiceAccountToken": false,
          "containers": Array [
            Object {
              "image": "talis/resque-web:stable",
              "imagePullPolicy": "IfNotPresent",
              "name": "resque",
              "ports": Array [
                Object {
                  "containerPort": 3000,
                  "protocol": "TCP",
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "100m",
                  "memory": "200Mi",
                },
                "requests": Object {
                  "cpu": "50m",
                  "memory": "100Mi",
                },
              },
            },
          ],
          "priorityClassName": "web",
        },
      },
    },
  },
]
`;

exports[`ResqueWeb Customizations 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "annotations": Object {
        "talis.io/chat": "https://talis.slack.com/archives/C04P9DPCX",
        "talis.io/description": "Frontend to the Resque job queue system",
        "talis.io/eks-dashboard": "None",
        "talis.io/graphs": "None",
        "talis.io/incidents": "None",
        "talis.io/issues": "https://github.com/talis/platform/issues",
        "talis.io/logs": "None",
        "talis.io/repository": "https://github.com/talis/resque-web-container",
        "talis.io/runbook": "None",
        "talis.io/uptime": "None",
        "talis.io/url": "https://resque-web.example.com",
      },
      "labels": Object {
        "app": "resque",
        "instance": "resque-web",
        "release": "latest",
        "role": "server",
      },
      "name": "test-resque-web-resque-web-service-c81bf306",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 3000,
          "protocol": "TCP",
          "targetPort": 3000,
        },
      ],
      "selector": Object {
        "app": "resque",
        "instance": "resque-web",
        "role": "server",
      },
      "type": "NodePort",
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1",
    "kind": "Ingress",
    "metadata": Object {
      "annotations": Object {
        "alb.ingress.kubernetes.io/listen-ports": "[{\\"HTTP\\":80},{\\"HTTPS\\":443}]",
        "alb.ingress.kubernetes.io/load-balancer-attributes": "idle_timeout.timeout_seconds=60",
        "alb.ingress.kubernetes.io/load-balancer-name": "resque-web",
        "alb.ingress.kubernetes.io/ssl-policy": "ELBSecurityPolicy-TLS-1-2-2017-01",
        "alb.ingress.kubernetes.io/ssl-redirect": "443",
        "alb.ingress.kubernetes.io/success-codes": "200,303",
        "alb.ingress.kubernetes.io/tags": "service=resque,instance=resque-web",
        "alb.ingress.kubernetes.io/target-type": "instance",
      },
      "labels": Object {
        "app": "resque",
        "instance": "resque-web",
        "release": "latest",
        "role": "server",
      },
      "name": "test-resque-web-resque-web-ingress-c8c81984",
    },
    "spec": Object {
      "defaultBackend": Object {
        "service": Object {
          "name": "test-resque-web-resque-web-service-c81bf306",
          "port": Object {
            "number": 3000,
          },
        },
      },
      "ingressClassName": "aws-load-balancer-internal",
      "tls": Array [
        Object {
          "hosts": Array [
            "*.example.com",
          ],
        },
      ],
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "resque",
        "instance": "resque-web",
        "release": "latest",
        "role": "server",
      },
      "name": "test-resque-web-c82b282b",
    },
    "spec": Object {
      "replicas": 1,
      "revisionHistoryLimit": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "resque",
          "instance": "resque-web",
          "role": "server",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "resque",
            "instance": "resque-web",
            "release": "latest",
            "role": "server",
          },
        },
        "spec": Object {
          "affinity": Object {
            "podAntiAffinity": Object {
              "preferredDuringSchedulingIgnoredDuringExecution": Array [
                Object {
                  "podAffinityTerm": Object {
                    "labelSelector": Object {
                      "matchLabels": Object {
                        "app": "resque",
                        "instance": "resque-web",
                        "role": "server",
                      },
                    },
                    "topologyKey": "topology.kubernetes.io/zone",
                  },
                  "weight": 100,
                },
              ],
            },
          },
          "automountServiceAccountToken": false,
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "RAILS_ENV",
                  "value": "production",
                },
                Object {
                  "name": "RAILS_RESQUE_REDIS",
                  "value": "redis:6379",
                },
                Object {
                  "name": "SECRET_KEY_BASE",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "SECRET_KEY_BASE",
                      "name": "resque-web-secrets",
                    },
                  },
                },
              ],
              "image": "talis/resque-web:latest",
              "imagePullPolicy": "IfNotPresent",
              "name": "resque",
              "ports": Array [
                Object {
                  "containerPort": 3000,
                  "protocol": "TCP",
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "100m",
                  "memory": "200Mi",
                },
                "requests": Object {
                  "cpu": "50m",
                  "memory": "100Mi",
                },
              },
            },
          ],
          "priorityClassName": "web",
        },
      },
    },
  },
]
`;
